ml-proto/spec/int.ml

```diff
@@ -208,7 +208,35 @@ struct
   let ge_s x y = x >= y
   let ge_u x y = cmp_u x (>=) y
 
-  let of_string = Rep.of_string
+  (* to_string supports unsigned decimals *)
+  let of_string x =
+    let unsigned_decimal x =
+      let ten = Rep.of_int 10 in
+      let l = String.length x in
+      (* First overflow-check via String.length *)
+      if l > (if Rep.bitwidth = 32 then 10 else 20)
```

Comment:
Hm, this effectively hard-codes that we only instantiate with 32 and 64 bit. It would be more modular to have a generic criterium, e.g. `bitwidth / 3`.